src/backend/backend_concept.h

```diff
@@ -15,20 +17,29 @@ namespace snmalloc
    * get_metadata takes a bool-ean template parameter indicating whether it may
    * be accessing memory that is not known to be committed.
    */
-  template<typename Meta>
+  template<
+    typename Meta,
+    typename BackendMetadata = std::remove_pointer_t<
+      decltype(std::declval<Meta>()
+                 .template get_metaentry_mut<true>(std::declval<address_t>())
+                 .get_meta())>>
   concept ConceptBackendMeta =
-    requires(
-      address_t addr,
-      size_t sz,
-      const MetaEntry& t)
+    requires(address_t addr, size_t sz, const MetaEntry<BackendMetadata>& t)
   {
-    { Meta::set_metaentry(addr, sz, t) } -> ConceptSame<void>;
+    {
+      Meta::set_metaentry(addr, sz, t)
```

Comment:
set_metaentry is not used by the frontend anymore, so it could be removed from the concept. Right @mjp41 ?

Reply:
Gone.